lib/src/chatbox.dart

```diff
@@ -96,6 +96,7 @@ class ChatBox extends StatefulWidget {
   final TranslateConversations? translateConversations;
   final List<String> highlightedWords;
   final BaseMessagePredicate? messageFilter;
+  final String? scrollToMessage;
```

Comment:
drive-by semantics nitpick: it's weird to me that something with a verb is a property and not a method. notably, what does it mean when i change it?

eg what about:
- make a chatbox with scrollToMessage=foo
- users scrolls around
- now the code wants to scroll to the same message again
- setting scrollToMessage=foo won't work because it's already set to that

and also, setting it to `null`, will that also affect scrolling? 

concretely, is there a downside if this were a method? note, i dont know much dart or flutter, i don't even know whether methods can be a thing in flutter components

Reply:
I can reply to some of your points:

Reply:
> It's very unusual for a Flutter component to have methods

I would say Flutter Widgets aren't designed to have methods that can be called externally. A common workaround that people use to store and access a reference to Widgets is via [GlobaKeys](https://api.flutter.dev/flutter/foundation/Key-class.html). Keys function in a similar vein as they do in React. So using them as a way to store an object's reference so that you can use its methods is not intended.

Franco's suggestion of having a `ChatBoxController` is the next best option.

Reply:
We could have a `onLoadEnd` property that takes a function which will receive the Controller as an argument.

Reply:
right, i see. if i may ask, why did you merge this with a pending discussion?

i was going to suggest naming it differently, eg scrollToMessageAtLoad or startAtMessage or something like that, to underline that it only determines the _initial_ message being scrolled into view and doesn't otherwise control the scroll position. not sure if that discussion still has merit now that it got merged though